File: app/src/main/java/eu/darken/bluemusic/main/core/service/modules/MusicVolumeModule.java
package eu.darken.bluemusic.main.core.service.modules;

import javax.inject.Inject;

import eu.darken.bluemusic.main.core.audio.AudioStream;
import eu.darken.bluemusic.main.core.audio.StreamHelper;
import eu.darken.bluemusic.main.core.service.BlueMusicServiceComponent;
import eu.darken.bluemusic.settings.core.Settings;

@BlueMusicServiceComponent.Scope
public class MusicVolumeModule extends BaseVolumeModule {

    @Inject
    public MusicVolumeModule(Settings settings, StreamHelper streamHelper) {
        super(settings, streamHelper);
    }

    @Override
    AudioStream.Type getType() {
        return AudioStream.Type.MUSIC;
    }
}

File: app/src/main/java/eu/darken/bluemusic/main/core/service/modules/events/MusicVolumeModule.kt
package eu.darken.bluemusic.main.core.service.modules.events

import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import dagger.multibindings.IntoSet
import eu.darken.bluemusic.main.core.audio.AudioStream
import eu.darken.bluemusic.main.core.audio.StreamHelper
import eu.darken.bluemusic.main.core.service.modules.EventModule
import eu.darken.bluemusic.main.core.Settings
import javax.inject.Inject

import javax.inject.Singleton

@Singleton
class MusicVolumeModule @Inject constructor(
        settings: Settings, streamHelper: StreamHelper
) : BaseVolumeModule(settings, streamHelper) {

    override val type: AudioStream.Type = AudioStream.Type.MUSIC

    @Module @InstallIn(SingletonComponent::class)
    abstract class Mod {
        @Binds @IntoSet abstract fun bind(entry: MusicVolumeModule): EventModule
    }
}